libkbfs/folder_block_ops.go

```diff
@@ -1417,7 +1624,9 @@ func (fbo *folderBlockOps) writeDataLocked(
 	// the `deCache` is used to determine whether there are any dirty
 	// files.  TODO: combine `deCache` with `dirtyFiles` and
 	// `unrefCache`.
-	fbo.deCache[file.tailPointer().Ref()] = newDe
+	cacheEntry := fbo.deCache[file.tailPointer().Ref()]
+	cacheEntry.dirEntry = newDe
```

Comment:
If it wasn't in the cache before, then cacheEntry (which is a non-pointer type) is the struct zero value, with all the fields at zero values.  So this new code should be exactly equivalent to the old code.